### PR TITLE
Fix GLOBAL IN table queries with analyzer.

### LINKED ISSUE #62286: new analyzer: global in TABLE does not work in distributed queries
I cannot reproduce it with fiddle https://fiddle.clickhouse.com/5de17519-8a1b-4cfc-8e5d-2e76cd7b94cf

It's reproducible only with 2 real shards:

```sql
drop table if exists u;
drop table if exists ff1 on cluster cluster2shards;
drop table if exists d on cluster cluster2shards;

CREATE TABLE ff1 on cluster cluster2shards (uid Int16) ENGINE = MergeTree order by tuple() as select 1;

create table d on cluster cluster2shards as ff1 Engine=Distributed(cluster2shards, 'default', 'ff1');

CREATE TABLE u(uid Int16) ENGINE=Memory as select 1;

select * from d where uid GLOBAL IN (select * from u);
   ┌─uid─┐
1. │   1 │
   └─────┘
   ┌─uid─┐
2. │   1 │
   └─────┘

select * from d where uid GLOBAL IN u settings allow_experimental_analyzer=0;
   ┌─uid─┐
1. │   1 │
   └─────┘
   ┌─uid─┐
2. │   1 │
   └─────┘

select * from d where uid GLOBAL IN u settings allow_experimental_analyzer=1;
   ┌─uid─┐
1. │   1 │
   └─────┘
↙ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.)
1 row in set. Elapsed: 0.005 sec.

Received exception from server (version 24.3.2):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Received from host2:9000. DB::Exception: 
Unknown expression or table expression identifier 'default.u' in scope SELECT __table1.uid AS uid 
FROM default.ff1 AS __table1 WHERE __table1.uid GLOBAL IN (default.u AS __table2). (UNKNOWN_IDENTIFIER)
```

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `Unknown expression or table expression identifier` error for `GLOBAL IN table` queries (with new analyzer). Fixes #62286

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit bcccbe2b19bd50429ce6de1f4e0619c0677fa4a1 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62409/bcccbe2b19bd50429ce6de1f4e0619c0677fa4a1/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Planner/CollectSets.cpp [hunk 1]
```diff
@@ -24,6 +24,34 @@ namespace ErrorCodes
     extern const int UNSUPPORTED_METHOD;
 }
 
+QueryTreeNodePtr makeExecutableSubqueryForIn(const QueryTreeNodePtr & in_second_argument, const ContextPtr & context)
```
Comment (kitaisreal) [inline src/Planner/CollectSets.cpp:27]: This method needs to be at the end of this file, because it is last in the interface file.

Comment (kitaisreal) [inline src/Planner/CollectSets.cpp:27]: Please reuse `Planner/Utils.h` `buildSubqueryToReadColumnsFromTableExpression`.

Comment (kitaisreal) [reply src/Planner/CollectSets.cpp:27]: Also it is better to move this function to `Planner/Utils.h`.

**Review by kitaisreal — CHANGES_REQUESTED** [review]

**Pull request merged**